episodes/02-data-structures.Rmd

```diff
@@ -0,0 +1,216 @@
+---
+title: 'Data Structures'
+teaching: 10
+exercises: 2
+---
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- What are the basic data types in R?
+- How do I represent categorical information in R?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- To be aware of the different types of data.
+- To begin exploring data frames, and understand how they are related to vectors, factors and lists.
+- To be able to ask questions from R about the type, class, and structure of an object.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Vectors 
+So far we've looked on individual values. Now we will move to a data structure 
+called vectors. Vectors are arrays of values of a same data type. 
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: callout
+
+### Data types 
+
+Data type refers to a type of information that is stored by a value. 
+It can be:
+
+- `numerical` (a number)
+- `integer` (a number without information about decimal points)
+- `logical` (a boolean - are values TRUE or FALSE?)
+- `character` (a text/ string of characters)
+- `complex` (a complex number)
+- `raw` (raw bytes)
+
+We won't discuss `complex` or `raw` data type in the workshop.
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+You can create a vector with a `c()` function. 
+
+```{r vectors}
+
+numeric_vector <- c(2, 6, 3) # vector of numbers - numeric data type.
+numeric_vector
+
+character_vector <- c('banana', 'apple', 'orange') # vector of words - or strings of characters- character data type
+character_vector
+
+logical_vector <- c(TRUE, FALSE, TRUE) # vector of logical values (is something true or false?)- logical data type.
+logical_vector
+
+```
+
+### Combining vectors 
+
+The combine function, `c()`, will also append things to an existing vector:
+
+```{r combine-vectors}
+
+ab_vector <- c('a', 'b')
+ab_vector
+
+abcd_vector <- c(ab_vector, 'c', 'd')
+abcd_vector
+
+```
+
+### Missing values  
+
+A common operation you want to perform is to remove all the missing values 
+(in R denoted as `NA`). Let's have a look how to do it: 
+
```

Comment:
Let the participants create the vector as a mini-challenge in the sake of interactivity?

Reply:
ok

Reply:
@KyriJanssen, @fcjerome, can you please update the text with a challenge. Again, you can do that as a suggestion using this button:
![image](https://github.com/carpentries-incubator/r-geospatial-urban/assets/33600128/a60b7e23-7cac-4a33-bfc9-e0238a589b9a)

You can select multiple lines before making suggestions. See how it's done here: https://stackoverflow.com/questions/48911087/how-to-select-comment-on-a-range-of-lines-in-github-pull-request